client/index.js

```diff
@@ -37,3 +37,8 @@ class AppProvider extends React.Component {
 //addAllStyles();
 
 render(<AppProvider />, document.getElementById('app'));
+if (process.env.NODE_ENV === 'development' && module.hot) {
```

Comment:
Checking `module.hot` is enough here I think. When HMR is not wanted then `HotModuleReplacementPlugin` shouldn't be included by webpack. `process.env.NODE_ENV` is more for the server side.